theopensystemslab/planx-new
# feat: add basic API endpoint to copy an internal portal and transform into own flow

Changes:
- Adds an endpoint that accepts a `flowId` and `portalNodeId`, and extracts the internal portal out of the parent flow transforming it into it's own flow with unique node ids.

Other cleanup (adds extra noise, sorry, but hopefully worth it!):
- Adds an `/editor` subdirectory in the API so we can better organize endpoints used in the Editor only
- Updates legacy files to reference the new, common GraphQL client definitions

To test:
- Login to pizza (so your JWT gets set)
- Try https://api.1146.planx.pizza/flows/86d80cbe-04ad-4cec-aaba-4d1833f19033/copy-portal/sldSDLTujR (this is the Permitted Development example from the attached Trello card)
- Confirm it returns 200, paste `"data"` result into a flow you manually create in pizza and confirm there are no corrupted nodes
- It should match this original portal https://editor.planx.uk/opensystemslab/permitteddevelopment,sldSDLTujR
